### PR TITLE
Add a basic configuration file support

### PR DESCRIPTION
**Pull request by gaetanlhf**
It is now possible to define the library path and the port to use from a configuration file. :)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -60,9 +71,9 @@ func startServer(zimService *zim.Handler) {
 }
 
 // Build a new Http server
-func createServer(router http.Handler) http.Server {
+func createServer(router http.Handler, port string) http.Server {
 	return http.Server{
-		Addr:    ":8080",
+		Addr:    port,
```
Comment (JojiiOfficial) [inline main.go]: I think you should rename `port` to address. ":8080" is just a shortcut for "0.0.0.0:8080" which is the full address where the server is going to listen on, so "address" as config and variable name would be more fitting.

File: main.go [hunk 2]
```diff
@@ -11,12 +11,23 @@ import (
...
 
 func main() {
 	setupLogger()
 
-	libPath := "./library"
+	config, err := toml.LoadFile("config.toml")
+
+	// If the configuration file does not exist, return an error
+	if err != nil {
+		log.Error(err)
```
Comment (JojiiOfficial) [inline main.go]: `log.Error(err)` doesn't really stop the program, it just prints out the error and continues. Maybe a `return` or even better a default config file generation at this point would make more sense?

File: README.md [hunk 3]
```diff
@@ -32,13 +32,22 @@ A modern zim fileserver which can handle multiple zim files by serving a beautif
...
+```toml
+[Config]
+  LibraryPath = "./library"
+  Port = ":8080"
+```
+
 # Usage
-Create a folder `library` and place your .zim files inside it, or link them using symlinks.<br>
-Run the binary and go to `https://localhost:8080`
+Your `LibraryPath` must contain your .zim files inside it.
```
Comment (JojiiOfficial) [inline README.md]: Why did you remove the `symlink` hint?

File: main.go [hunk 4]
```diff
@@ -40,12 +51,12 @@ func main() {
 		return
 	}
 
-	startServer(service)
+	startServer(service, port)
 }
 
-func startServer(zimService *zim.Handler) {
+func startServer(zimService *zim.Handler, port string) {
```
Comment (JojiiOfficial) [inline main.go]: Maybe creating a config structure and passing this one instead would make it easier to add more config features later on?

File: README.md (hunk 3 above)
Comment (gaetanlhf) [reply README.md]: Sorry, I forgot to put this part in. :(
Which is a bit of a pity, since it is a feature that is presented further down in the README...
This is fixed in c2dbfac.

File: main.go (hunk 2 above)
Comment (gaetanlhf) [reply main.go]: En effet, je ne sais pas pourquoi je n'ai pas ajouté une configuration par défaut...
C'est tellement logique de faire ça comme ça.
Please check 31436ea and a4b146c. :)

Comment (gaetanlhf) [reply main.go]: Indeed, I don't know why I didn't add a default configuration...
It's so logical to do it like that in fact.
Please check 31436ea and a4b146c. :)

File: main.go (hunk 1 above)
Comment (gaetanlhf) [reply main.go]: I named it `Port` because it seemed simple to understand for some users, but yes, it is indeed a good idea to change that.
Please check 31436ea and a4b146c. :)

File: main.go (hunk 4 above)
Comment (gaetanlhf) [reply main.go]: I hope that I have read your mind well what you were expecting. :)

**Comment by gaetanlhf** [comment]
The latest changes have arrived and are ready for your comments. :)

File: README.md [hunk 5]
```diff
@@ -32,13 +32,22 @@ A modern zim fileserver which can handle multiple zim files by serving a beautif
...
 or
 - Download the latest release
 
+# Configuration
+### Example
+config.toml:
+```toml
+[Config]
+  LibraryPath = "./library"
+  Address = ":8080"
```
Comment (JojiiOfficial) [inline README.md]: Are those spaces added on purpose? Afaik is toml not indented with spaces at all

Comment (gaetanlhf) [reply README.md]: Indentation seems to be allowed but not required.
Since you're making the point, I guess you prefer I take them off?

Comment (JojiiOfficial) [reply README.md]: Yes, I think toml looks cleaner this way.

Comment (gaetanlhf) [reply README.md]: Okay, in that case, it's corrected.

**Comment by JojiiOfficial** [comment]
Thank you!
